plugins/package-managers/swiftpm/src/main/kotlin/SwiftPmModel.kt

```diff
@@ -70,6 +71,23 @@ data class PinV1(
     )
 }
 
+/**
+ * See https://github.com/apple/swift-package-manager/blob/3ef830dddff459e569d6e49c186c3ded33c39bcc/Sources/PackageGraph/PinsStore.swift#L387-L462.
+ */
+@Serializable
+data class PinV2(
+    val identity: String,
+    val state: PinV1.State?,
+    val location: String
+) {
+    @Serializable
+    data class State(
+        val version: String? = null,
+        val revision: String? = null,
+        val branch: String? = null
+    )
```

Comment:
Remove this if indeed `PinV1.State` should be used. Also, consider unnested the state class from `PinV1` then to better explain that it's shared.

Reply:
I've changed it according to above comment to use PinV2.State to keep things independent.